Update ANGLE to 2023-10-14 (cd6b265c262346dca0c236b9bcc99f403a43197c)

kkinnunen-apple opened the pull request:
#### 3b928ce58d577d94a25fe6fba82be98a7638f3fb
<pre>
Update ANGLE to 2023-10-14 (cd6b265c262346dca0c236b9bcc99f403a43197c)
<a href="https://bugs.webkit.org/show_bug.cgi?id=263188">https://bugs.webkit.org/show_bug.cgi?id=263188</a>
rdar://problem/117004308

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 3d75b794df0faee30b82ce9ef52b590ebf3511ae..cd6b265c262346dca0c236b9bcc99f403a43197c --pretty=%h %s
cd6b265c26 Roll VK-GL-CTS from 7a1ab4d3eb20 to 7d738783bf28 (1 revision)
6154bd93b4 Remove binding from DescriptorInfoDesc.
21ea0a3c85 Add gzip tests when hitting a decompression error
93781cd328 Roll VK-GL-CTS from f2434516dcf5 to 7a1ab4d3eb20 (8 revisions)
7660855415 Vulkan: use cpu transcoding for small texture size.
1988108e60 Tighten draw buffers validation for dual-source blending
b870956ee6 GL: Limit depth/stencil texture mode updates
ad1cc5bfc6 Roll vulkan-deps from 37653f7c343e to f719b699697b (12 revisions)
481f381de8 Roll Chromium from 137cc90d5084 to 345cb7cebed5 (603 revisions)
5b7763f9d4 Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize
10d4d0ae4e Revert &quot;Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize&quot;
3938a005e8 Roll vulkan-deps from 1d5140e06b47 to 37653f7c343e (17 revisions)
e4e499b7ff Roll Chromium from f9691c6faf3c to 137cc90d5084 (563 revisions)
24dabdbbee Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize
90dd58a22b Vulkan: Reduce mGarbageMutex lock contention
2608c622a3 Vulkan: Refactor SharedGarbageList into templated class
7f3af6d2f2 Roll vulkan-deps from be2755d4d109 to 1d5140e06b47 (13 revisions)
0935ab3f35 Roll Chromium from 2f77720808c4 to f9691c6faf3c (638 revisions)
ea332c7155 Manual roll VK-GL-CTS from a55b0930e9db to f2434516dcf5 (7 revisions)
43acadabfd Add missing configurations for khr tests
6f794eab7c Change angle::FixedQueue&apos;s storage from std::array to std::vector
05c21cfc68 ANGLE Tracer: Make GL API capture threadsafe
53e37a3eef Replace mActiveTextures.fill(nullptr) with memset
a0967770ba Add egl/wrapper headers to deqp opengl sources
d1cf2d4326 Cleanup sampler completeness validation
b4d87668a8 Roll vulkan-deps from d3dc8dfdccca to be2755d4d109 (8 revisions)
04ce14af00 Roll Chromium from b3c67dea8ed9 to 2f77720808c4 (504 revisions)
99b208c1e6 Roll vulkan-deps from f770c9936a5a to d3dc8dfdccca (57 revisions)
bc959555a9 Roll Chromium from 4622adaa4bf3 to b3c67dea8ed9 (723 revisions)
0afc3ec954 Metal: Fix robust initialization of PVRTC1 textures
7e8907db8a Add egl/wrapper to deqp opengl includes
0015d1e213 Allow ETC1_RGB8 for 2D array textures
5149d21009 Vulkan: Deduplicate sync&apos;s clientWait logic
adb17c8018 Vulkan: Copy 3D&lt;-&gt;2DArray images with vkCmdBlitImage anyway
65e8ae30e1 Vulkan: ANGLE_device_vulkan is incompatible w/ asyncCommandQueue
d6bf8e0df0 Improve capture_replay_tests (win-trace) logging
3671dbbd58 Roll Chromium from 138350604230 to 4622adaa4bf3 (433 revisions)
8c341cfdef Vulkan: Fix blits involving 3D images
48474d15cd Manual roll Chromium from 07002c74826e to 138350604230 (1338 revisions)
856a6ec06d Enable rgb-565-no-depth-no-stencil tests on Android Bots

Canonical link: <a href="https://commits.webkit.org/269357@main">https://commits.webkit.org/269357@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c4aadcbf3be5ff71f890b279315e05db5c504020

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22349 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22547 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24251 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20684 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26810 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22880 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21694 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22586 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/22281 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19401 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25106 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/19331 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20256 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26503 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20348 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20489 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24363 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/21019 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17810 "Found 1 new test failure: media/video-remove-insert-repaints.html (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/20474 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5317 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24708 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/21012 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/ThirdParty/ANGLE/.gitmodules:
```diff
@@ -1,11 +1,47 @@
 	path = third_party/clang-format/script
 	url = https://chromium.googlesource.com/external/github.com/llvm/llvm-project/clang/tools/clang-format.git
 	gclient-condition = not build_with_chromium
+[submodule "testing"]
+	path = testing
+	url = https://chromium.googlesource.com/chromium/src/testing
+	gclient-condition = not build_with_chromium
+[submodule "third_party/abseil-cpp"]
+	path = third_party/abseil-cpp
+	url = https://chromium.googlesource.com/chromium/src/third_party/abseil-cpp
+	gclient-condition = not build_with_chromium
+[submodule "third_party/android_build_tools"]
+	path = third_party/android_build_tools
+	url = https://chromium.googlesource.com/chromium/src/third_party/android_build_tools
+	gclient-condition = checkout_android and not build_with_chromium
+[submodule "third_party/android_deps"]
+	path = third_party/android_deps
+	url = https://chromium.googlesource.com/chromium/src/third_party/android_deps
+	gclient-condition = checkout_android and not build_with_chromium
+[submodule "third_party/android_platform"]
+	path = third_party/android_platform
+	url = https://chromium.googlesource.com/chromium/src/third_party/android_platform
+	gclient-condition = checkout_android and not build_with_chromium
+[submodule "third_party/android_sdk"]
+	path = third_party/android_sdk
+	url = https://chromium.googlesource.com/chromium/src/third_party/android_sdk
+	gclient-condition = checkout_android and not build_with_chromium
```
Did you trim these out before? I reverted the additions in my change to remove them. Maybe that's what caused the issues with my previous roll.

kkinnunen-apple replied:
I haven't treated the file any different to other files, consciously.

This time, I imported the upstream version of the file for this update, in order to remove all our "changes" to it.

djg reviewed (approved):
:shipit:

kkinnunen-apple committed:
Update ANGLE to 2023-10-14 (cd6b265c262346dca0c236b9bcc99f403a43197c)
https://bugs.webkit.org/show_bug.cgi?id=263188
rdar://problem/117004308

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 3d75b794df0faee30b82ce9ef52b590ebf3511ae..cd6b265c262346dca0c236b9bcc99f403a43197c --pretty=%h %s
cd6b265c26 Roll VK-GL-CTS from 7a1ab4d3eb20 to 7d738783bf28 (1 revision)
6154bd93b4 Remove binding from DescriptorInfoDesc.
21ea0a3c85 Add gzip tests when hitting a decompression error
93781cd328 Roll VK-GL-CTS from f2434516dcf5 to 7a1ab4d3eb20 (8 revisions)
7660855415 Vulkan: use cpu transcoding for small texture size.
1988108e60 Tighten draw buffers validation for dual-source blending
b870956ee6 GL: Limit depth/stencil texture mode updates
ad1cc5bfc6 Roll vulkan-deps from 37653f7c343e to f719b699697b (12 revisions)
481f381de8 Roll Chromium from 137cc90d5084 to 345cb7cebed5 (603 revisions)
5b7763f9d4 Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize
10d4d0ae4e Revert "Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize"
3938a005e8 Roll vulkan-deps from 1d5140e06b47 to 37653f7c343e (17 revisions)
e4e499b7ff Roll Chromium from f9691c6faf3c to 137cc90d5084 (563 revisions)
24dabdbbee Fix tsan failure of test FixedQueue.ConcurrentPushPopWithResize
90dd58a22b Vulkan: Reduce mGarbageMutex lock contention
2608c622a3 Vulkan: Refactor SharedGarbageList into templated class
7f3af6d2f2 Roll vulkan-deps from be2755d4d109 to 1d5140e06b47 (13 revisions)
0935ab3f35 Roll Chromium from 2f77720808c4 to f9691c6faf3c (638 revisions)
ea332c7155 Manual roll VK-GL-CTS from a55b0930e9db to f2434516dcf5 (7 revisions)
43acadabfd Add missing configurations for khr tests
6f794eab7c Change angle::FixedQueue's storage from std::array to std::vector
05c21cfc68 ANGLE Tracer: Make GL API capture threadsafe
53e37a3eef Replace mActiveTextures.fill(nullptr) with memset
a0967770ba Add egl/wrapper headers to deqp opengl sources
d1cf2d4326 Cleanup sampler completeness validation
b4d87668a8 Roll vulkan-deps from d3dc8dfdccca to be2755d4d109 (8 revisions)
04ce14af00 Roll Chromium from b3c67dea8ed9 to 2f77720808c4 (504 revisions)
99b208c1e6 Roll vulkan-deps from f770c9936a5a to d3dc8dfdccca (57 revisions)
bc959555a9 Roll Chromium from 4622adaa4bf3 to b3c67dea8ed9 (723 revisions)
0afc3ec954 Metal: Fix robust initialization of PVRTC1 textures
7e8907db8a Add egl/wrapper to deqp opengl includes
0015d1e213 Allow ETC1_RGB8 for 2D array textures
5149d21009 Vulkan: Deduplicate sync's clientWait logic
adb17c8018 Vulkan: Copy 3D<->2DArray images with vkCmdBlitImage anyway
65e8ae30e1 Vulkan: ANGLE_device_vulkan is incompatible w/ asyncCommandQueue
d6bf8e0df0 Improve capture_replay_tests (win-trace) logging
3671dbbd58 Roll Chromium from 138350604230 to 4622adaa4bf3 (433 revisions)
8c341cfdef Vulkan: Fix blits involving 3D images
48474d15cd Manual roll Chromium from 07002c74826e to 138350604230 (1338 revisions)
856a6ec06d Enable rgb-565-no-depth-no-stencil tests on Android Bots

Canonical link: https://commits.webkit.org/269357@main
